packages/lab/src/date-input/DateInput.tsx

```diff
@@ -192,27 +308,70 @@ export const DateInput = forwardRef<HTMLDivElement, DateInputProps>(
           },
           className
         )}
-        ref={ref}
+        onClick={(event) => handleInputClick(event)}
+        ref={inputRef}
         {...rest}
       >
         <input
           aria-describedby={clsx(formFieldDescribedBy, dateInputDescribedBy)}
           aria-labelledby={clsx(formFieldLabelledBy, dateInputLabelledBy)}
-          className={clsx(withBaseName("input"), inputProps?.className)}
+          className={withBaseName("input")}
           disabled={isDisabled}
           readOnly={isReadOnly}
-          ref={inputRef}
-          role={role}
+          ref={startInputRef}
           tabIndex={isDisabled ? -1 : 0}
           onBlur={handleStartDateBlur}
           onChange={handleStartDateChange}
+          onKeyDown={handleStartDateKeyDown}
           onFocus={!isDisabled ? handleFocus : undefined}
           placeholder={placeholder}
-          value={value}
-          {...restA11yProps}
+          size={placeholder.length}
+          value={
+            isReadOnly && !startDateStringValue
+              ? emptyReadOnlyMarker
+              : startDateStringValue
+          }
           {...restDateInputProps}
           required={isRequired}
         />
+        {selectionVariant === "range" && (
+          <>
+            <span>-</span>
+            <input
+              aria-describedby={clsx(
+                formFieldDescribedBy,
+                dateInputDescribedBy
+              )}
+              aria-labelledby={clsx(formFieldLabelledBy, dateInputLabelledBy)}
+              className={withBaseName("input")}
+              disabled={isDisabled}
+              readOnly={isReadOnly}
+              ref={endInputRef}
+              tabIndex={isDisabled ? -1 : 0}
+              onBlur={handleEndDateBlur}
+              onChange={handleEndDateChange}
+              onKeyDown={handleEndDateKeyDown}
+              onFocus={!isDisabled ? handleFocus : undefined}
+              placeholder={placeholder}
+              size={placeholder.length}
+              value={
+                isReadOnly && !endDateStringValue
+                  ? emptyReadOnlyMarker
+                  : endDateStringValue
+              }
+              {...restDateInputProps}
+              required={isRequired}
```

Comment:
A little random that `required` is after `rest`. If this is intentional, add some comment.